### PR TITLE
feat(user): use github username in ranking

### LINKED ISSUE #112: Use github user name instead of real name in ranking
At least the firebase auth provides the screenName, but there seems not be any public api to get this.
screen shot 2017-04-15 at 14 05 58

### PR DESCRIPTION
**Pull request by hisabimbola**
The first draft of my solution added some comments in the code

resolves #112

### CODE REVIEW

File: src/api/app.js [hunk 1]
```diff
@@ -12,6 +24,17 @@ export default {
     }
     xhr.send()
   },
+  getGithubUsername: function (email, callback) {
+    const url = `https://api.github.com/search/commits?q=author-email:${email}&sort=author-date&per_page=1`
```
Comment (samiheikki) [inline src/api/app.js:28]: Does the `api` have some daily limits?

File: src/api/app.js [hunk 2]
```diff
@@ -12,6 +24,17 @@ export default {
...
+    const headers = [{
+      key: 'Accept',
+      value: 'application/vnd.github.cloak-preview'
+      // TODO add token
+    }]
+    this.getJSON(url, headers, function (err, response) {
```
Comment (samiheikki) [inline src/api/app.js:34]: You can change this to arrow function

File: src/api/app.js [hunk 3]
```diff
@@ -47,5 +70,13 @@ export default {
       logos[optionNumbers[2]],
       logos[optionNumbers[3]]
     ]
+  },
+  isValidArray: function (value) {
+    if (!Array.isArray) {
```
Comment (samiheikki) [inline src/api/app.js:75]: `isArray` support seems good enough so I don't think it needs to be polyfilled.

File: src/api/app.js [hunk 4]
```diff
@@ -12,6 +24,17 @@ export default {
     }
     xhr.send()
   },
+  getGithubUsername: function (email, callback) {
+    const url = `https://api.github.com/search/commits?q=author-email:${email}&sort=author-date&per_page=1`
+    const headers = [{
+      key: 'Accept',
+      value: 'application/vnd.github.cloak-preview'
+      // TODO add token
```
Comment (samiheikki) [inline src/api/app.js:32]: It needs a token?

File: src/api/app.js (hunk 1 above)
Comment (hisabimbola) [reply src/api/app.js:28]: Yes, Github has a rate limit of 5000 requests per hour as seen [here ](https://developer.github.com/apps/building-integrations/setting-up-and-registering-github-apps/about-rate-limits-for-github-apps/).

If this site has that much traffic then we can improve it a bit by adding token else no cause of alarm.

File: src/api/app.js (hunk 3 above)
Comment (hisabimbola) [reply src/api/app.js:75]: Okay, I was thinking of browser support but yeah I will change it.

File: src/api/app.js (hunk 4 above)
Comment (hisabimbola) [reply src/api/app.js:32]: We do not "need" one but good if we can add one.

One funny thing I noticed is that when the user is authenticated via OAuth2, we have the Github token that we can use for each user to authenticate the request to fetch Github username but sadly we only have that token when the user signs in, once the user leaves the app and come back, firebase does not return the token in the [onAuthStateChanged](https://github.com/samiheikki/javascript-guessing-game/pull/282/files#diff-ea994a51b0d9884f4fc046eca11a5c99R80) method; in fact, it does not persist it at all. So we probably have to persist the token ourselves when the user signs in.

But personally, I am not sure there is much need for that much logic. I think this works fine for the app for now

**Pull request closed without merging**